### PR TITLE
Bug 1179415 - Incorrect passcode now clears automatically R=Justin Wi…

### PR DESCRIPTION
**Pull request by jgw96**
…llis

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f0022bc32c4750bae704f4a3d0458299f73e25b1

File: apps/settings/js/panels/screen_lock_passcode/panel.js [hunk 1]
```diff
@@ -196,6 +196,9 @@ define(function(require) {
                   this._passcodeBuffer = '';
                   this._updatePassCodeUI();
                   this._showDialogInMode('new');
+                } else if(result === false) {
```
Comment (gasolin) [inline apps/settings/js/panels/screen_lock_passcode/panel.js]: `result === false` is equivalent to origin else case. Could you check again if it works by put `updatePassCodeUI` in the `else` case?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4c30b9e5438941a9bd8e17c80c540dcb5b8c6c4f

File: apps/settings/js/panels/screen_lock_passcode/panel.js (hunk 1 above)
Comment (jgw96) [inline apps/settings/js/panels/screen_lock_passcode/panel.js]: This is changed in the latest commit

File: apps/settings/js/panels/screen_lock_passcode/panel.js [hunk 2]
```diff
@@ -196,8 +198,10 @@ define(function(require) {
                   this._passcodeBuffer = '';
                   this._updatePassCodeUI();
                   this._showDialogInMode('new');
+                  //below are my changes
```
Comment (gasolin) [inline apps/settings/js/panels/screen_lock_passcode/panel.js]: please remove the redundant comment

File: apps/settings/js/panels/screen_lock_passcode/panel.js [hunk 3]
```diff
@@ -187,6 +188,7 @@ define(function(require) {
                   });
                 } else {
                   this._passcodeBuffer = '';
+                  this.updatePassCodeUI();
```
Comment (gasolin) [inline apps/settings/js/panels/screen_lock_passcode/panel.js]: `_updatePassCodeUI` ?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=364534b2f5c0b5859e1dc680a5151252238ce45d

File: apps/settings/js/panels/screen_lock_passcode/panel.js (hunk 3 above)
Comment (jgw96) [inline apps/settings/js/panels/screen_lock_passcode/panel.js]: Sorry not used to that style of naming methods, fixed in my last commit

**Comment by jgw96** [comment]
will now squash commits

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b78afdd6ee5f36a8e9c7be428537c77e53bafa1a

**Comment by gasolin** [comment]
I think you forgot handle the case 'new' that has 2 passcode fields

**Pull request closed without merging**